lib/compute/index.js

```diff
@@ -0,0 +1,1053 @@
+ * @private
+ */
+var Network = require('./network.js');
+
+/**
+ * @type {module:compute/operation}
+ * @private
+ */
+var Operation = require('./operation.js');
+
+/**
+ * @type {module:compute/region}
+ * @private
+ */
+var Region = require('./region.js');
+
+/**
+ * @type {module:compute/snapshot}
+ * @private
+ */
+var Snapshot = require('./snapshot.js');
+
+/**
+ * @type {module:common/streamrouter}
+ * @private
+ */
+var streamRouter = require('../common/stream-router.js');
+
+/**
+ * @type {module:common/util}
+ * @private
+ */
+var util = require('../common/util.js');
+
+/**
+ * @type {module:compute/zone}
+ * @private
+ */
+var Zone = require('./zone.js');
+
+/**
+ * @const {string}
+ * @private
+ */
+var COMPUTE_BASE_URL = 'https://www.googleapis.com/compute/v1/projects/';
+
+/**
+ * Required scopes for Google Compute Engine API.
+ * @const {array}
+ * @private
+ */
+var SCOPES = ['https://www.googleapis.com/auth/compute'];
+
+/**
+ * A Compute object allows you to interact with the Google Compute Engine API.
+ * Using this object, you can access your instances with
+ * {module:compute/instance}, disks with {module:compute/disk}, and firewalls
+ * with {module:compute/firewall}.
+ *
+ * @alias module:compute
+ * @constructor
+ *
+ * @param {object} options - [Configuration object](#/docs/?method=gcloud).
+ *
+ * @example
+ * var gcloud = require('gcloud')({
+ *   keyFilename: '/path/to/keyfile.json',
+ *   projectId: 'grape-spaceship-123'
+ * });
+ *
+ * var compute = gcloud.compute();
+ */
+function Compute(options) {
+  if (!(this instanceof Compute)) {
+    return new Compute(options);
+  }
+
+  options = options || {};
+
+  if (!options.projectId) {
+    throw util.missingProjectIdError;
+  }
+
+  var authConfig = {
+    credentials: options.credentials,
+    keyFile: options.keyFilename,
+    scopes: SCOPES,
+    email: options.email
+  };
+
+  this.authConfig = authConfig;
+
+  this.makeAuthorizedRequest_ = util.makeAuthorizedRequestFactory(authConfig);
+
+  this.projectId = options.projectId;
+}
+
+/**
+ * Create a firewall. For a detailed description of method's options see
+ * [API reference](https://goo.gl/kTMHep).
+ *
+ * @param {string} name - Name of the firewall.
+ * @param {object} config - Configuration object. See a
+ *     [Firewall resource](https://goo.gl/7FpjXA) for detailed information.
+ * @param {function} callback - The callback function.
+ *
+ * @example
+ * var config = {
+ *   allowed: [
+ *     {
+ *       IPProtocol: 'tcp',
+ *       ports: ['3000']
+ *     }
+ *   ],
+ *   sourceRanges: ['0.0.0.0/0']
+ * };
+ *
+ * function callback(err, firewall, operation, apiResponse) {
+ *   // `firewall` is a Firewall object.
+ *
+ *   // `operation` is an Operation object that can be used to check the status
+ *   // of the request.
+ * }
+ *
+ * compute.createFirewall('new-firewall-name', config, callback);
+ */
+Compute.prototype.createFirewall = function(name, config, callback) {
```

Comment:
Oh and this is `(name, config, callback)` instead of `(config, callback)` just to feel more consistent with our API, eg `storage.createBucket(name, options, callback)`